crates/revm/revm-inspectors/src/tracing/js/builtins.rs

```diff
@@ -122,6 +129,92 @@ pub(crate) fn to_bigint(value: U256, ctx: &mut Context<'_>) -> JsResult<JsValue>
         ctx,
     )
 }
+/// Takes three arguments: a JavaScript value that represents the sender's address, a string salt
+/// value, and the initcode for the contract. Compute the address of a contract created by the
+/// sender with the given salt and code hash, then converts the resulting address back into a byte
+/// buffer for output.
+pub(crate) fn to_contract2(
+    _: &JsValue,
+    args: &[JsValue],
+    ctx: &mut Context<'_>,
+) -> JsResult<JsValue> {
+    // Extract the sender's address, salt and initcode from the arguments
+    let from = args.get_or_undefined(0).clone();
+    let salt = match args.get_or_undefined(1).to_string(ctx) {
+        Ok(js_string) => {
+            let s = js_string.to_std_string().unwrap();
+            match hex::decode(s) {
+                Ok(bytes) => H256::from_slice(bytes.as_slice()),
+                Err(_) => {
+                    return Err(JsError::from_native(
+                        JsNativeError::typ().with_message("invalid salt hex string"),
+                    ))
+                }
+            }
+        }
+        Err(_) => {
+            return Err(JsError::from_native(JsNativeError::typ().with_message("invalid salt type")))
+        }
+    };
```

Comment:
I think we already have this:

https://github.com/paradigmxyz/reth/blob/4d3d4974a48b4e838ccda50c3171da11c4885a13/crates/revm/revm-inspectors/src/tracing/js/builtins.rs#L133-L147

I believe this is equivalent